scm/driver/bitbucket/linker.go

```diff
@@ -24,8 +25,19 @@ func (l *linker) Resource(ctx context.Context, repo string, ref scm.Reference) (
 	case scm.IsPullRequest(ref.Path):
 		d := scm.ExtractPullRequest(ref.Path)
 		return fmt.Sprintf("%s%s/pull-requests/%d", l.base, repo, d), nil
-	case ref.Sha == "":
+	case scm.IsBranch(ref.Path) && ref.Sha == "":
 		t := scm.TrimRef(ref.Path)
+
+		// Bitbucket has a bug where the "source view" link for
+		// a branch which contains a slash results in a 404.
+		// The link to the "branch view" works with names containing
+		// a slash so we do this for branches with slashes in its names.
+		// See https://jira.atlassian.com/browse/BCLOUD-14422 for more information
+		//
+		if strings.Contains(t, "/") {
```

Comment:
one concern I have is that we could introduce a regression where the sha is empty, but the object type is not a branch. Could we adjust accordingly:

```diff
-case scm.IsBranch(ref.Path) && ref.Sha == "":
+ref.Sha == "":
```

```diff
-if strings.Contains(t, "/") {
+if scm.IsBranch(ref.Path) && strings.Contains(t, "/") {
```